main.cpp

```diff
@@ -50,7 +50,28 @@ struct Wrapper
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
+    }
+    void print() const
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+private:
+    Type val;
+};
+
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& t) : val(std::move(t))
+    {
+        std::cout << "Wrapper(Point&&)" << std::endl;
+    }
+    void print() const
+    {
+        std::cout << val.toString() << std::endl;
+    }
+private:
+    Point val;
```

Comment:
Project 9.1 Challenge 1:
There is a lot of repetition between the template class `Wrapper`, and the template specialization for `Wrapper<Point>`
Can you figure out how to eliminate the duplicate constructor?
hint: you only have to _explicitly specialize_ that which is unique.
You can specialize member functions without specializing the entire class :wink:
- [ ] revise